
```{r setup}
library(data.table)
library(plink2R)
knitr::opts_knit$set(root.dir = '~/Documents/GitHub/dsc_190_final/')
```
# Step 0: Preprocessing Data

preprocssing.R

# Step 1: Baseline Model

-   Fusion command per gene basis:
      - Identify gene number per gene basis (1-50)
      - Need bed, bim, fam for each gene
        - hm3 snps
        - cis -> 500 kb
      - need to define a temp file path for intermediate files
  
- PLINK command to create bed, bim, fams:

BASE_DIR=~/Documents/Github/dsc_190_final

files=$(ls ${BASE_DIR}/extract/)
for file in $files; do
    # Extract chromosome number between chr_ and _.txt
    chr_num=$(echo $file | grep -o 'chr_[0-9]*' | grep -o '[0-9]*')
    
    ./plink \
    --bfile ${BASE_DIR}/eur_1000G/1000G_eur_chr${chr_num} \
    --extract ${BASE_DIR}/extract/${file} \
    --make-bed \
    --out ${BASE_DIR}/gene_filtered_1kg/${file}_1kg_chr${chr_num}
done


- Fusion command:

files=$(cat ../fusion_file.txt)
for file in $files; do
  Rscript FUSION.compute_weights.R \
    --bfile ../gene_filtered_1kg/${file} \
    --tmp ../int_fusion \
    --out fusion_basemodel/${file} \
    --models enet \
    --PATH_plink "~/Desktop/UCSD_Senior/Fall/DSC_190/plink"
done



```{r baseline}
ge <- fread('~/Documents/GitHub/dsc_190_final/filtered_geuvadis_dataset.tsv')
ge <- ge[order(as.numeric(ge$Chr)), ]
chr <- 1
hm3snps <- fread("~/Documents/GitHub/dsc_190_final/w_hm3.snplist", header = T)

gt <- read_plink(paste0('~/Documents/GitHub/dsc_190_final/eur_1000G/1000G_eur_chr', 1))
for (i in 1:50) {
  upper <- ge$Coord[i] + 500000
  lower <- ge$Coord[i] - 500000
  if (as.numeric(ge$Chr)[i] != chr) {
    chr <- as.numeric(ge$Chr)[i]
    gt <- read_plink(paste0('~/Documents/GitHub/dsc_190_final/eur_1000G/1000G_eur_chr', chr))
  }
  in_range <- which(gt$bim$V4 >= lower & gt$bim$V4 <= upper)
  cis_snps <- gt$bim$V2[in_range]
  
  hm3_filt <- which(cis_snps %in% hm3snps$SNP)
  cis_snps <- cis_snps[hm3_filt]
  
  write.table(cis_snps, file = paste0("extract/extract_gene_", i, "_chr_", chr, "_.txt"), 
              quote = F, row.names = F, col.names = F)
}


```

# Step 2: Create a model for trans qtls

Same commands as above



```{f trans_qtls}
ge <- fread('~/Documents/GitHub/dsc_190_final/filtered_geuvadis_dataset.tsv')
ge <- ge[order(as.numeric(ge$Chr)), ]
chr <- 1
hm3snps <- fread("~/Documents/GitHub/dsc_190_final/w_hm3.snplist", header = T)

gt <- read_plink(paste0('~/Documents/GitHub/dsc_190_final/eur_1000G/1000G_eur_chr', 1))
for (i in 1:50) {
  upper <- ge$Coord[i] + 500000
  lower <- ge$Coord[i] - 500000
  if (as.numeric(ge$Chr)[i] != chr) {
    chr <- as.numeric(ge$Chr)[i]
    gt <- read_plink(paste0('~/Documents/GitHub/dsc_190_final/eur_1000G/1000G_eur_chr', chr))
  }
  in_range <- which(gt$bim$V4 <= lower & gt$bim$V4 >= upper)
  cis_snps <- gt$bim$V2[in_range]
  
  hm3_filt <- which(cis_snps %in% hm3snps$SNP)
  cis_snps <- cis_snps[hm3_filt]
  
  write.table(cis_snps, file = paste0("extract_trans/extract_gene_", i, "_chr_", chr, "_.txt"), 
              quote = F, row.names = F, col.names = F)
}

```

# Step 3: Processing Epigenetic Data

```{r epigenetic}
enhancer <- fread('~/Documents/GitHub/dsc_190_final/B_cell_blood_enhancers.bed.txt')
filt <- which(enhancer$V1 != 'chrX' & enhancer$V1 != 'chrY')
enhancer<- enhancer[filt, ]


atac <- fread('~/Documents/GitHub/dsc_190_final/B_cell_atac.bed.gz')
filt <- which(atac$V1 != 'chrX' & atac$V1 != 'chrY')
atac <- atac[filt, ]

hic_domains <- fread('~/Documents/GitHub/dsc_190_final/B_cell_contact_domains.bedpe.gz')
colnames(hic_domains)[1] <- "V1"
filt <- which(hic_domains$V1 != 'chrX' & hic_domains$V1 != 'chrY')
hic_domains[filt, ]

hic_loops <- fread('~/Documents/GitHub/dsc_190_final/B_cell_loops.bedpe.gz')
colnames(hic_loops)[1] <- "V1"
filt <- which(hic_loops$V1 != 'chrX' & hic_loops$V1 != 'chrY')
hic_loops[filt, ]

```
# Step 4: Creating Bins 

Intersect bed files

000:

001:

bedtools intersect -a B_cell_contact_domains.bedpe.gz \
-b B_cell_atac_bed.gz B_cell_blood_enhancers.bed.txt \
-v > 001.bed


010:

bedtools intersect -a B_cell_atac_bed.gz \
-b B_cell_contact_domains.bedpe.gz B_cell_blood_enhancer.bed.txt \
-v > 010.bed

100:

bedtools intersect -a B_cell_blood_enhancer.bed.txt \ 
-b B_cell_contact_domains.bedpe.gz B_cell_atac_bed.gz \
-v > 100.bed

011:

bedtools intersect -a B_cell_atac.bed.gz \
-b B_cell_contact_domains.bedpe.gz > 011_int.bed

bedtools intersect -a 011_int.bed \
-b B_cell_enhancers.bed.txt \
-v > 011.bed

110:

bedtools intersect -a B_cell_blood_enhancer.bed.txt \
-b B_cell_atact.bed.gz> 110_int.bed

bedtools intersect -a 110_int.bed \
-b B_cell_contact_domains.bedpe.gz \
-v > 110.bed

101:

bedtools intersect -a B_cell_blood_enhancer.bed.txt \
-b B_cell_contact_domains.bedpe.gz > 101_int.bed

bedtools intersect -a 101_int.bed \
-b B_cell_atac.bed.gz \
-v > 101.bed

111: 

bedtools intersect -a B_cell_blood_enhancer.bed.txt \
-b B_cell_contact_domains.bedpe.gz B_cell_atac.bed.gz > 111.bed



```{r }


```




